src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,429 @@
+@[simp] lemma nat_cast_snd (n : ℕ) : (n : 𝓜(𝕜 , A)).snd = n := rfl
+@[simp] lemma int_cast_fst (n : ℤ) : (n : 𝓜(𝕜 , A)).fst = n := rfl
+@[simp] lemma int_cast_snd (n : ℤ) : (n : 𝓜(𝕜 , A)).snd = n := rfl
+@[simp] lemma pow_fst (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).fst = a.fst ^ n := rfl
+@[simp] lemma pow_snd (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).snd = a.snd ^ n := rfl
+
+/-- The natural injection from `double_centralizer.to_prod` except the second coordinate inherits
+`mul_opposite.op`. The ring structure on `𝓜(𝕜, A)` is the pullback under this map. -/
+def to_prod_mop : 𝓜(𝕜, A) → ((A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ) :=
+λ a, (a.fst, mul_opposite.op a.snd)
+
+/-- The ring structure is inherited as the pullback under the injective map
+`double_centralizer.to_prod_mop : 𝓜(𝕜, A) → (A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ` -/
+instance : ring 𝓜(𝕜, A) :=
+function.injective.ring to_prod_mop
+  (λ a b h, let h' := prod.ext_iff.mp h in ext _ _ $ prod.ext h'.1 $ mul_opposite.op_injective h'.2)
+  rfl rfl (λ _ _, rfl) (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl)
+  (λ x n, by simpa only [to_prod_mop, to_prod, prod.smul_fst, prod.smul_snd, mul_opposite.op_smul])
+  (λ x n, by simpa only [to_prod_mop, to_prod, prod.smul_fst, prod.smul_snd, mul_opposite.op_smul])
+  (λ x n, by simpa only [to_prod_mop, to_prod, prod.pow_fst, prod.pow_fst, mul_opposite.op_pow])
+  (λ _, rfl) (λ _, rfl)
+
+/-- The canonical map `double_centralizer.to_prod` as an additive group homomorphism. -/
+def add_group_hom_to_prod : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A) :=
+{ to_fun := to_prod,
+  map_zero' := rfl,
+  map_add' := λ x y, rfl }
+
+/-- The module structure is inherited as the pullback under the additive group monomoprhism
+`double_centralizer.to_prod : 𝓜(𝕜, A) →+ (A →L[𝕜] A) × (A →L[𝕜] A)` -/
+instance : module 𝕜 𝓜(𝕜, A) :=
+function.injective.module 𝕜 add_group_hom_to_prod ext (λ x y, rfl)
+
+noncomputable instance : algebra 𝕜 𝓜(𝕜, A) :=
+algebra.of_module
+  (λ k a b, by {ext; simp only [mul_fst, smul_fst, mul_snd, smul_snd, coe_smul',pi.smul_apply,
+    continuous_linear_map.coe_mul, function.comp_app, continuous_linear_map.map_smul]})
+  (λ k a b, by {ext; simp only [mul_fst, smul_fst, mul_snd, smul_snd, algebra.mul_smul_comm,
+    coe_smul', continuous_linear_map.coe_mul, pi.smul_apply, function.comp_app]})
+
+/-!
+### Star structure
+-/
+
+section star
+
+variables [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A]
+
+/-- The star operation on `a : 𝓜(𝕜, A)` is given by
+`(star a).to_prod = (star ∘ a.snd ∘ star, star ∘ a.fst ∘ star)`. -/
+instance : has_star 𝓜(𝕜, A) :=
+{ star := λ a,
+  { fst := (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.snd).comp
+      ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A),
+    snd := (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.fst).comp
+      ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A),
+    central := λ x y, by simpa only [star_mul, star_star]
+      using (congr_arg star (a.central (star y) (star x))).symm } }
```

Comment:
Maybe you could define more generally `module (A ≃L[𝕜] A) 𝓜(𝕜, A)`? Then this would be
```suggestion
/-- The star operation on `a : 𝓜(𝕜, A)` is given by
`(star a).to_prod = (star ∘ a.snd ∘ star, star ∘ a.fst ∘ star)`. -/
instance : has_star 𝓜(𝕜, A) :=
{ star := (\smul) ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A ≃L[𝕜] A) }
```

Reply:
1. I can't imagine we would actually want this as an instance, but more importantly ...
2. I don't think this works because you would need the continuous semilinear (because you need to apply it to `≃ₗᵢ⋆`) equiv to *also* be antimultiplicative in order to prove the centrality condition. This is somehow rather specific to `star`.